CV_Analyzer_clean_v2.m

```diff
@@ -0,0 +1,101 @@
+% INTRODUCTION
+% 1. Please put the '.mpt' files you want to analyze in the same folder as
+% this file.
+% 2. Update the mass and molecular weight
+% 3. Click 'Run'
+% 4. You should expect a text file that contains the analyzed outcome
+
+% mass = 0.49*(105.53*0.933/(105.53*0.933+20.36+1056.28*0.05))/1000; %mass of the electrode (unit: g)
+mass = 1;
+MW = 265.81; % molecular weight of your sample 
+
+% Load files
+allFiles = dir( '*.mpt' );% get the information all .mpt file in the current folder
+allNames = {allFiles(~[allFiles.isdir]).name};% get the names of the .mpt files
+Num_El = numel(allNames);% count the number of files in the folder
+Ewe = NaN(Num_El,1);% create a matrix to contain working electrode potential
+I = NaN(Num_El,1);% create a matrix to contain current
+Cath_Cap = NaN(Num_El,1);% create a matrix to contain cathodic capacity
+Anod_Cap = NaN(Num_El,1);% create a matrix to contain anodic capacity
+sweep_rates = NaN(Num_El,1);% create a matrix to sweep rates
+ for i = 1:1:Num_El% a for loop function to extract data from each .mpt file
+     % To find out headerlines and sweep rate
+    textFileName = char(allNames(i)); % convert the name of the i-th file into a string format readable by Matlab
+    fileID = fopen(textFileName,'rt');% open the i-th file for getting headerline and sweep rate infos
+    D = textscan(fileID,'%s','Delimiter','\n');% scan the text in the i-th file and save to D
+    fclose(fileID);% close the file
+    Headerlines_Num = str2num(D{1}{2}(19))*10+str2num(D{1}{2}(20));% Find the number of headerlines in the 2nd row of D{1} which are stored in the 19th and 20th positions and convert the string format to a number format
+        % find the sweep rate
+    for sweep_rate_finder = 1:Headerlines_Num
+        Sweep_Rate_Line = D{1}{sweep_rate_finder};
+        Sweep_Rate_Key = 'dE/dt               ';
+        Sweep_Rate_Index = strfind(Sweep_Rate_Line, Sweep_Rate_Key);
+        if Sweep_Rate_Index == 1
+            sweep_rate(i) = sscanf(Sweep_Rate_Line(Sweep_Rate_Index(1) + length(Sweep_Rate_Key):end), '%g', 1);% scan the line that stores the sweep rate data from the position that the Sweep_Rate_Key ends to where the line ends. The number is stored in the i-th position in the sweep_rate matrix.
+        end
+    end
+    % to read the columns
+    fileID = fopen(textFileName,'rt');% open the i-th file for getting actual data
+    C = textscan(fileID,'%d %d %d %d %d %f %f %f %f %d %f %f','Headerlines',Headerlines_Num);% obtain data
```

Comment:
I remember this has occasionally caused issues since we manually enter the text types for each column. Is there a way to automatically handle this in case someone's file looks slightly different? Otherwise we can add a note about this in the README file with instructions.